File: cs/Minsk/CodeAnalysis/Binding/BoundBinaryExpression.cs
namespace Minsk.CodeAnalysis.Binding;

internal class BoundBinaryExpression : BoundExpression
{
    public BoundBinaryExpression(BoundExpression left, BoundBinaryOperator op, BoundExpression right)
    {
        Left = left;
        Op = op;
        Right = right;
    }

    public BoundExpression Left { get; }
    public BoundBinaryOperator Op { get; }
    public BoundExpression Right { get; }
    public override Type Type => Op.ResultType;
    public override BoundNodeKind Kind => BoundNodeKind.BinaryExpression;
}

File: kt/Minsk/minsk/src/main/kotlin/dev/phytolizer/minsk/analysis/binding/BoundBinaryExpression.kt
package dev.phytolizer.minsk.analysis.binding

import java.lang.reflect.Type

internal class BoundBinaryExpression(
    val left: BoundExpression,
    val op: BoundBinaryOperator,
    val right: BoundExpression,
) : BoundExpression() {
    override val type: Type
        get() = op.resultType
    override val kind: BoundNodeKind
        get() = BoundNodeKind.BinaryExpression
}
